### ISSUE TITLE
Default values at the client side are fetched when App is fresh installed

### ISSUE THREAD

**Issue by muhammad-hassan-shakeel** [launchdarkly/flutter-client-sdk#51]
**Describe the bug**
When the app is fresh installed, default values at the client side is read instead of values on the launch darkly's actual value. Then after I kill the app and launch it again. It works fine

**To reproduce**
Delete the previously installed app and then rerun the app

**Expected behavior**
Actual values should be read from LaunchDarkly servers.

**SDK version**
Flutter (Channel stable, 3.10.6, on macOS 13.4.1
    22F770820d darwin-arm64, locale en-GB)
launchdarkly_flutter_client_sdk: ^2.0.3

**Language version, developer tools**
Flutter (Channel stable, 3.10.6, on macOS 13.4.1
    22F770820d darwin-arm64, locale en-GB)

**OS/platform**
Android 13

**Comment by tanderson-ld** [comment]
Hi @mHassan20896 , could you provide a snippet of your startup code?  Not enough information has been provided to indicate there is an issue at the moment with the SDK itself.  Perhaps there is and we'll diagnose.

The SDK will take some non-zero amount of time to get the latest flag data from the server.  If there is no internet, it may not even be possible, so we must use an approach that lends itself to these unavoidable cases.

There are a few approaches that can be taken.
1. Launching into the content of the application and having a UX that facilitates reactive flag updates.  This is recommended and can be achieved [with listeners](https://pub.dev/documentation/launchdarkly_flutter_client_sdk/latest/launchdarkly_flutter_client_sdk/LDClient/registerFeatureFlagListener.html).

2. Having a UX that gets flag values frequently (think every time a user navigates to a screen).  This is not recommended for new projects as it will be difficult to become reactive in the future, which is where the industry is trending.

3. Waiting on startup for the latest flag values to be retrieved before letting the user interact with content.  This can be achieved with the [startFuture method].(https://pub.dev/documentation/launchdarkly_flutter_client_sdk/latest/launchdarkly_flutter_client_sdk/LDClient/startFuture.html)

**Comment by mHassan20896** [comment]
```
LDConfig config =
        LDConfigBuilder(environmentDetails.launchDarklyMobileKey).build();

    _builder
        .kind(kind, userKey)
        .anonymous(true);

    LDContext context = _builder.build();

    await LDClient.startWithContext(config, context);
    await LDClient.startFuture();
    await _fetchFlags();
```

@tanderson-ld  I am calling these methods right inside main.dart

**Comment by t-unit** [comment]
@tanderson-ld any update on this issue? It's blocking us from updating from version 2.1.0 on one of our projects

**Comment by tanderson-ld** [comment]
@t-unit, we are currently working on a rewrite of our Flutter SDK to use Dart and are looking to address this issue in this rewrite.  It may be a couple months away from release but we are aiming for very early next year.

**Comment by t-unit** [comment]
@tanderson-ld Any news about when we can expect a new version that will fix our issues?

**Comment by tanderson-ld** [comment]
We have just released `launchdarkly_flutter_client_sdk: 4.0.0-alpha.0` if you would like to be an early tester.

The example below will wait up to 5 seconds on fresh install for data to be fetched.  After the internal cache for a context is situated, subsequent startups and calls to `start()` may complete even before the latest flags are fetched by using the cached flag values.

await client.start().timeout(const Duration(seconds: 5));

**Comment by t-unit** [comment]
@tanderson-ld thanks for the info, we will give the new version a try
